### PR TITLE
fix(ext/crypto) - exportKey JWK for AES/HMAC must use base64url

### PR DESCRIPTION
**Pull request by SeanWykes**
Solves #13263 by refactoring JWK symmetric-key export (AES/HMAC) to rust, which already uses base64url.

### CODE REVIEW

**Review by littledivy — APPROVED** [review]
LGTM except one request below

File: ext/crypto/00_crypto.js [hunk 1]
```diff
@@ -3057,11 +3053,18 @@
...
           kty: "oct",
-          k: unpaddedBase64(innerKey.data),
         };
+
+        // 3.
+        const data = core.opSync("op_crypto_export_key", {
+          format: "jwksecret",
+          algorithm: key[_algorithm].name,
+        }, innerKey);
+        ObjectAssign(jwk, data);
```
[the remark this answers is not in the document]
Comment (SeanWykes) [reply ext/crypto/00_crypto.js:3066]: No real reason, just staying in line with other export code.

File: cli/tests/unit/webcrypto_test.ts [hunk 2]
```diff
@@ -1310,3 +1310,28 @@ Deno.test(async function testImportEcSpkiPkcs8() {
     assertEquals(new Uint8Array(expPrivateKeySPKI), spki);*/
   }
 });
+
+Deno.test(async function testHMACJwkBase64Url() {
```
[the remark this answers is not in the document]
Comment (SeanWykes) [reply cli/tests/unit/webcrypto_test.ts:1314]: Will do. Things are getting complicated without merging some of other crypto PRs ;)
Maybe also add test for AES export (same rust, different js path).

[the remark this answers is not in the document]
Comment (SeanWykes) [reply cli/tests/unit/webcrypto_test.ts]: done, added new tests

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @SeanWykes!

**Comment by SeanWykes** [comment]
@littledivy @bartlomieju is there any reason this merge is failing CI on Deno main branch?

**Comment by bartlomieju** [comment]
@SeanWykes we had an outage that caused several CI failures.

**Pull request merged**